sdk/servicebus/azure-servicebus/api_samples.py

```diff
@@ -0,0 +1,219 @@
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        batch_sequence_numbers = topic_sender.schedule(batch_schedule_message, schedule_time_utc)
+
+        # 1.4 cancel schedule
+        topic_sender.cancel_scheduled_messages(sequence_number)
+        topic_sender.cancel_scheduled_messages(batch_sequence_numbers)
+
+# 2 Receive
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+queue_receiver = sb_client.get_queue_receiver(queue_name="queue_name")
+
+# 2.1 peek
+with sb_client:
+    with queue_receiver:
+        msgs = queue_receiver.peek(count=10)
+        for msg in msgs:
+            print(msg)
+
+# 2.2 iterator receive
+with sb_client:
+    with queue_receiver:
+        for msg in queue_receiver:
+            print(message)
+            msg.complete()
+            # msg.renew_lock()
+            # msg.abondon()
+            # msg.defer()
+            # msg.dead_letter()
+
+# 2.3 pull mode receive
+with sb_client:
+    with queue_receiver:
+        msgs = queue_receiver.receive(max_batch_size=10)
+        for msg in msgs:
+            msg.complete()
+
+# 2.4 receive deferred letter
+with sb_client:
+    with queue_receiver:
+        defered_sequence_numbers = [1,2,3,4,5,6]
+        deferred_messages = queue_receiver.receive_deferred_messages(defered_sequence_numbers)
+        for i in deferred_messages:
+            deferred_messages.abandon()
+
+
+### Session operation
+# 1 Send
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+topic_sender = sb_client.get_topic_sender(topic_name="topic_name")
+
+with sb_client:
+    with topic_sender:
+        # 1.1 send single
+        message = Message("Test message")
+        topic_sender.send(message, session_id="test_session")
+
+        # 1.2 send batch
+        batch_message = topic_sender.create_batch()
+        while True:
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        topic_sender.send(batch_message, session_id="test_session")
+
+        # 1.3 schedule
+        schedule_message = Message("Scheduled message")
+        schedule_time_utc = datetime.utcnow() + timedelta(minutes=10)
+        sequence_number = topic_sender.schedule(schedule_message, schedule_time_utc, session_id="test_session")
+
+        batch_schedule_message = topic_sender.create_batch()
+
+        batch_sequence_numbers = topic_sender.schedule(
+            message=batch_schedule_message,
+            schedule_time_utc=schedule_time_utc,
+            session_id="test_session"
+        )
+
+        # 1.4 cancel schedule
+        topic_sender.cancel_scheduled_messages(sequence_number)
+        topic_sender.cancel_scheduled_messages(batch_sequence_numbers)
+
+# 2 Receive
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+subscription_receiver = sb_client.get_subscription_receiver(
```

Comment:
Same issue in our JS API and I believe we can fix it in the same way (make the return be a Promise, rather than just the object).

Now with that said - @JoshLove-msft is there any reason to have session vs non-session behave differently? Should they both just actually start/attempt to start the link when you call 'get*Receiver'?

Reply:
They certainly could both do that, but then we should probably also do it for sending. The reason for only adding it for sessions is that the get next session case is the only case where a failure to open the link is expected. All other cases should eventually open the link just fine.